src/style/style.js

```diff
@@ -1120,6 +1120,7 @@ class Style extends Evented {
         for (const sourceCache of sourceCaches) {
             sourceCache.setFeatureState(sourceLayer, target.id, state);
         }
+        this.fire(new Event('setfeaturestate'));
```

Comment:
The fix should also apply when `removeFeatureState` is called to clear a value. When you do that, maybe a different name is appropriate

Reply:
@karimnaaji Thoughts on how to get this behavior working now that this is no longer using events? Should `removeFeatureState` call a new function `tile.removeFeatureState` to trigger this update?

Reply:
I haven't stepped through the code in debug, but from reading it `removeFeatureState` should eventually result in a bucket update. I'd verify that with a step through to confirm.

If we tie a render cache invalidation to a buffer update, we ensure that when something actually changes visually (like you do now in this PR) we invalidate the draping cache entry. `bucket.update` should be the end result of any mutation of feature state changes, which is the end result of going through `source_state:coalesceChanges`.

But to confirm that some unit test confirming correct invalidation of these cases would be necessary.

Reply:
Debugging shows that 1source_state:coalesceChanges` (and thus the fix) is indeed being called after `removeFeatureState` through `.coalesceChanges()`. I'm not totally sure how that works, but I can confirm that it works as expected in browser testing.